<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>editorgeneralwidget</class>
 <widget class="QWidget" name="editorgeneralwidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>721</width>
    <height>982</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="grpIndents">
     <property name="title">
      <string>Indents</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="chkAutoIndent">
        <property name="text">
         <string>Auto Indent</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkTabToSpaces">
        <property name="text">
         <string>Replace tab with spaces</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_2" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="lbCaretColor_2">
           <property name="text">
            <string>Tab Width</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spTabWidth">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>16</number>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkShowIndentLines">
        <property name="text">
         <string>Show Indent Lines</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_3" native="true">
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="2">
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lbIndentLineColor">
           <property name="text">
            <string>Indent Line Color</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="ColorEdit" name="colorIndentLine">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkFillIndents">
        <property name="text">
         <string>Fill Indents</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="grpCarets">
     <property name="title">
      <string>Caret</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QCheckBox" name="chkEnhanceHome">
        <property name="text">
         <string>Move caret to the first non-space char in the current line when press HOME key</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkEnhanceEndKey">
        <property name="text">
         <string>Move caret to the last non-space char in the current line when press END key</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkKeepCaretX">
        <property name="text">
         <string>Keep X position of the caret when moving vertically</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget" native="true">
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="spacing">
          <number>7</number>
         </property>
         <item row="1" column="1">
          <widget class="QComboBox" name="cbCaretForOverwrite"/>
         </item>
         <item row="3" column="1">
          <widget class="ColorEdit" name="colorCaret">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lbCaretForInsert">
           <property name="text">
            <string>Caret for inserting mode</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="cbCaretForInsert"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="lbCaretColor">
           <property name="text">
            <string>Caret Color</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="lbCaretForOverwrite">
           <property name="text">
            <string>Caret for overwriting mode</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="3">
          <widget class="QCheckBox" name="chkCaretUseTextColor">
           <property name="text">
            <string>Use text color as caret color</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Scroll</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QCheckBox" name="chkAutoHideScrollBars">
        <property name="text">
         <string>Auto hide scroll bars</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkScrollPastEOL">
        <property name="text">
         <string>Can scroll the last char to the left edge of the editor</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkScrollPastEOF">
        <property name="text">
         <string>Can scroll the last line to the top edge of the editor</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkScrollHalfPage">
        <property name="text">
         <string>Page Up/Down scrolls half a page</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkScrollByOneLess">
        <property name="text">
         <string>Forces page scroll to be one line less</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_6" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Mouse Wheel Scroll Speed</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinMouseWheelScrollSpeed">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="grpRightEdge">
     <property name="title">
      <string>Show right edge line</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QWidget" name="widget_4" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="lbCaretColor_3">
           <property name="text">
            <string>Right egde width</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spRightEdge">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>80</number>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_5" native="true">
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="0" column="2">
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lbCaretColor_5">
           <property name="text">
            <string>Right edge line color</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="ColorEdit" name="colorRightEdgeLine">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ColorEdit</class>
   <extends>QFrame</extends>
   <header location="global">widgets/coloredit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
